module/executiondatasync/optimistic_syncing/core_impl_test.go

```diff
@@ -0,0 +1,360 @@
+	persistentRegisters := storagemock.NewRegisterIndex(t)
+	persistentEvents := storagemock.NewEvents(t)
+	persistentCollections := storagemock.NewCollections(t)
+	persistentTransactions := storagemock.NewTransactions(t)
+	persistentResults := storagemock.NewLightTransactionResults(t)
+	persistentTxResultErrMsg := storagemock.NewTransactionResultErrorMessages(t)
+
+	// Set up default expectations for persist operations
+	// These will be called by the real Persister during Persist()
+	persistentRegisters.On("Store", mock.Anything, mock.Anything).Return(nil).Maybe()
+	persistentEvents.On("BatchStore", mock.Anything, mock.Anything, mock.Anything).Return(nil).Maybe()
+	persistentCollections.On("BatchStoreLightAndIndexByTransaction", mock.Anything, mock.Anything).Return(nil).Maybe()
+	persistentTransactions.On("BatchStore", mock.Anything, mock.Anything).Return(nil).Maybe()
+	persistentResults.On("BatchStore", mock.Anything, mock.Anything, mock.Anything).Return(nil).Maybe()
+	persistentTxResultErrMsg.On("BatchStore", mock.Anything, mock.Anything, mock.Anything).Return(nil).Maybe()
+
+	core, err := NewCoreImpl(
+		logger,
+		executionResult,
+		header,
+		execDataRequester,
+		txResultErrMsgsRequester,
+		persistentRegisters,
+		persistentEvents,
+		persistentCollections,
+		persistentTransactions,
+		persistentResults,
+		persistentTxResultErrMsg,
+		config.db, // Use the configured DB
+	)
+	require.NoError(t, err)
+
+	return core, execDataRequester, txResultErrMsgsRequester
+}
+
+func assertCoreInitialized(t *testing.T, core *CoreImpl) {
+	require.NotNil(t, core)
+	assert.NotNil(t, core.execDataRequester)
+	assert.NotNil(t, core.txResultErrMsgsRequester)
+	assert.NotNil(t, core.executionResult)
+	assert.NotNil(t, core.header)
+	assert.NotNil(t, core.inmemRegisters)
+	assert.NotNil(t, core.inmemEvents)
+	assert.NotNil(t, core.inmemCollections)
+	assert.NotNil(t, core.inmemTransactions)
+	assert.NotNil(t, core.inmemResults)
+	assert.NotNil(t, core.inmemTxResultErrMsgs)
+	assert.NotNil(t, core.indexer)
+	assert.NotNil(t, core.persister)
+}
+
+// TestNewCoreImpl tests the successful creation of a CoreImpl instance.
+func TestNewCoreImpl(t *testing.T) {
+	core, _, _ := createTestCoreImpl(t)
+
+	assertCoreInitialized(t, core)
+}
+
+// TestCoreImpl_Download tests the Download method which retrieves execution data and transaction error messages.
+func TestCoreImpl_Download(t *testing.T) {
+	t.Run("successful download", func(t *testing.T) {
+		core, execDataRequester, txResultErrMsgsRequester := createTestCoreImpl(t)
+		ctx := context.Background()
+
+		expectedExecutionData := unittest.BlockExecutionDataFixture()
+		execDataRequester.On("RequestExecutionData", ctx).Return(expectedExecutionData, nil)
+
+		expectedTxResultErrMsgs := []flow.TransactionResultErrorMessage{
+			{
+				TransactionID: unittest.IdentifierFixture(),
+				Index:         0,
+				ErrorMessage:  "test error",
+				ExecutorID:    unittest.IdentifierFixture(),
+			},
+		}
```

Comment:
consider making a fixture function for this, either in this file, or in the unittests module with the others.